### ISSUE TITLE
Set mpirun to None when a single MPI task is being used (similar to handling of mpi-serial)

### ISSUE THREAD

**Issue by ekluzek** [ESMCI/cime#4619]
mpi-serial used to be required for building serially. It no longer is for modern MPI libraries.  One of the benefits of mpi-serial is that it removes the complexity of having to use a mpirun command and options which is both different on different machines, adds complexity, makes it harder to port to a machine for a serial case, and often means you can't run a simple case on the command line.

There is special handling in CIME for mpi-serial that allows mpirun to be None. For this I think similar logic can be used to set it to None, in env_mach_specific.py in _find_best_mpirun_match.

The other way to handle this would be to add a setting for NTASKS==1 to config_machine.xml for the specific machines we want to do this for.

**Comment by ekluzek** [comment]
Doing this allows users that are running serially (for example with CTSM single point tower sites such as NEON) to run short cases on login nodes with:

`./case.submit --no-batch`

without that it won't work on machines that don't allow mpirun to be used on the login nodes.

**Comment by jedwards4b** [comment]
@ekluzek I don't think we should encourage running on the login nodes.  Use qcmd or run on casper.

**Comment by ekluzek** [comment]
Good point @jedwards4b I've edited my above comment to emphasize the use on non-HPC machines. And give a specific warning for Derecho. Let me know if you have any further suggestions...

**Comment by ekluzek** [comment]
Note, the ability to run cases like this with

> ./case.submit --no-batch

is important for one user who has an unusual use-case. He runs a bunch of cases all at once (multi-instance doesn't work for him because the cases have differences that multi-instance doesn't allow), so it's under batch, but he needs the above to get his setup to work.
